tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -220,7 +220,81 @@ ROOT::Experimental::Detail::RPageSourceFile::RPageSourceFile(std::string_view nt
       *fMetrics.MakeCounter<RNTuplePlainCounter*> ("timeWallUnzip", "ns", "wall clock time spent decompressing"),
       *fMetrics.MakeCounter<RNTupleTickCounter<RNTupleAtomicCounter>*>("timeCpuRead", "ns", "CPU time spent reading"),
       *fMetrics.MakeCounter<RNTupleTickCounter<RNTuplePlainCounter>*> ("timeCpuUnzip", "ns",
-                                                                       "CPU time spent decompressing")
+                                                                       "CPU time spent decompressing"),
+      *fMetrics.MakeCounter<RNTupleCalcFloatPerf*> ("bwRead", "MiBi/s", "bandwidth compressed bytes read per second",
+         [](const RNTupleMetrics &metrics) -> std::pair<bool, double> {
+            if (const auto szReadPayload = metrics.GetCounter("RPageSourceFile.szReadPayload")) {
+               if (const auto szReadOverhead = metrics.GetCounter("RPageSourceFile.szReadOverhead")) {
+                  if (const auto timeWallRead = metrics.GetCounter("RPageSourceFile.timeWallRead")) {
+                     if (auto walltime = timeWallRead->GetValueAsInt()) {
+                        double payload = szReadPayload->GetValueAsInt();
+                        double overhead = szReadOverhead->GetValueAsInt();
+                        // unit: bytes / nanosecond = GB/s; MiBi = GB * 1000 / 1.024 / 1.024
+                        return {true, 1000. * (payload + overhead) / (1.024 * 1.024 * walltime)};
+                     }
+                  }
+               }
+            }
+            return {false, -1.};
+         }
+      ),
+      *fMetrics.MakeCounter<RNTupleCalcFloatPerf*> ("bwReadUnzip", "MiBi/s", "bandwidth uncompressed bytes read per second",
```

Comment:
Here I'm not sure what exactly we want to measure.  At the moment, the metric returns the same than `bwRead`.  We might want to show (uncompressed bytes / (time read + time decompress)).  We should keep in mind that

1. We don't necessarily decompress all the data that we read
2. Decompression and reading is overlapped, so we'd need a new counter that keeps track of the overall wall-clock time

Reply:
Sorry, that was just broken, I meant to divide unzipped bytes / read time. Please check again :-)